prefab_cloud_python/config_value_unwrapper.py

```diff
@@ -9,21 +13,53 @@ def __init__(self, type):
         super().__init__("Unknown config value type: %s" % type)
 
 
+class UnknownProvidedSourceException(Exception):
+    "Raised when a provided value has an unknown source"
+
+    def __init__(self, source):
+        super().__init__("Unknown provided source: %s" % source)
+
+
 class ConfigValueUnwrapper:
-    def unwrap(value, key, context=Context.get_current()):
-        if value is None:
+    def __init__(self, value, weighted_value_index=None):
+        self.value = value
+        self.weighted_value_index = weighted_value_index
+
+    def deepest_value(config_value, config_key, context=Context.get_current()):
+        if config_value and config_value.WhichOneof("type") == "weighted_values":
+            value, index = WeightedValueResolver(
+                config_value.weighted_values.weighted_values,
+                config_key,
+                context.get(config_value.weighted_values.hash_by_property_name),
+            ).resolve()
+            return ConfigValueUnwrapper(
+                ConfigValueUnwrapper.deepest_value(
+                    value.value, config_key, context
+                ).value,
+                index,
+            )
+        elif config_value and config_value.WhichOneof("type") == "provided":
+            if config_value.provided.source == Prefab.ProvidedSource.Value("ENV_VAR"):
+                raw = os.getenv(config_value.provided.lookup)
+                if raw is None:
+                    return ConfigValueUnwrapper(ConfigValueWrapper.wrap(""))
```

Comment:
adding a             LOG.warn "ENV Variable #{config_value.provided.lookup} not found. Using empty string."
would be nice.